### PR TITLE
Migrate to Compose V2 for dev branch

### PR DESCRIPTION
**Pull request by trnayan**
⚠️ Note on feature completeness ⚠️
From July 2023 Compose V1 stopped receiving updates. It's also no longer available in new releases of Docker Desktop. Compose V2 is included with all currently supported versions of Docker Desktop. For more information, see [Migrate to Compose V2](https://docs.docker.com/compose/migrate).

Compose V2, which was first released in 2020, is included with all currently supported versions of Docker Desktop. It offers an improved CLI experience, improved build performance with BuildKit, and continued new-feature development.

Description

How do I switch to Compose V2?
The easiest and recommended way is to make sure you have the latest version of Docker Desktop, which bundles the Docker Engine and Docker CLI platform including Compose V2. With Docker Desktop, Compose V2 is always accessible as docker compose.

### CODE REVIEW

File: dc-build.sh [hunk 1]
```diff
@@ -12,4 +12,9 @@ fi
 
 # Building images for all configurations
 # The docker build doesn't supply any environment variables to the Dockerfile, so we can use any profile.
-docker-compose --profile mysql-rabbitmq --profile postgres-redis --env-file ./docker/environments/postgres-redis.env build $1
+
+# From July 2023 Compose V1 stopped receiving updates. Reference: https://docs.docker.com/compose/reference/
+# docker-compose --profile mysql-rabbitmq --profile postgres-redis --env-file ./docker/environments/postgres-redis.env build $1
```
Comment (dwalleck) [inline dc-build.sh:17]: Minor nit, but couldn't this line be removed?

File: dc-down.sh [hunk 2]
```diff
@@ -12,4 +12,9 @@ fi
 
 # Stopping containers for all configurations
 # The environment must be provided but it doesn't make a difference which one
-docker-compose --profile mysql-rabbitmq --profile postgres-redis --env-file ./docker/environments/postgres-redis.env down $1
+
+# From July 2023 Compose V1 stopped receiving updates. Reference: https://docs.docker.com/compose/reference/
+# docker-compose --profile mysql-rabbitmq --profile postgres-redis --env-file ./docker/environments/postgres-redis.env down $1
```
Comment (dwalleck) [inline dc-down.sh:17]: Minor nit again, but we don't seem to need this line

File: dc-stop.sh [hunk 3]
```diff
@@ -12,4 +12,9 @@ fi
 
 # Stopping containers for all configurations
 # The environment must be provided but it doesn't make a difference which one
-docker-compose --profile mysql-rabbitmq --profile postgres-redis --env-file ./docker/environments/postgres-redis.env stop $1
+
+# From July 2023 Compose V1 stopped receiving updates. Reference: https://docs.docker.com/compose/reference/
+# docker-compose --profile mysql-rabbitmq --profile postgres-redis --env-file ./docker/environments/postgres-redis.env stop $1
```
Comment (dwalleck) [inline dc-stop.sh:17]: Same nit. I don't think we need this line

File: dc-unittest.sh [hunk 4]
```diff
@@ -73,4 +73,8 @@ then
 fi
 
 echo "Running docker compose unit tests with profile $PROFILE and test case $TEST_CASE ..."
-docker-compose --profile $PROFILE --env-file ./docker/environments/$PROFILE.env exec uwsgi bash -c "python manage.py test $TEST_CASE -v2 --keepdb"
+# From July 2023 Compose V1 stopped receiving updates. Reference: https://docs.docker.com/compose/reference/
+# docker-compose --profile $PROFILE --env-file ./docker/environments/$PROFILE.env exec uwsgi bash -c "python manage.py test $TEST_CASE -v2 --keepdb"
```
Comment (dwalleck) [inline dc-unittest.sh:77]: Same nit. I don't think we need this line

File: dc-up-d.sh [hunk 5]
```diff
@@ -27,4 +27,9 @@ else
 fi
 
 echo "Starting docker compose with profile $PROFILE in the background ..."
-docker-compose --profile $PROFILE --env-file ./docker/environments/$PROFILE.env up --no-deps -d
+
+# From July 2023 Compose V1 stopped receiving updates. Reference: https://docs.docker.com/compose/reference/
+# docker-compose --profile $PROFILE --env-file ./docker/environments/$PROFILE.env up --no-deps -d
```
Comment (dwalleck) [inline dc-up-d.sh:32]: Same nit. I don't think we need this line

File: dc-up.sh [hunk 6]
```diff
@@ -26,4 +26,9 @@ else
 fi
 
 echo "Starting docker compose with profile $PROFILE in the foreground ..."
-docker-compose --profile $PROFILE --env-file ./docker/environments/$PROFILE.env up --no-deps
+
+# From July 2023 Compose V1 stopped receiving updates. Reference: https://docs.docker.com/compose/reference/
+# docker-compose --profile $PROFILE --env-file ./docker/environments/$PROFILE.env up --no-deps
```
Comment (dwalleck) [inline dc-up.sh:31]: Same nit. I don't think we need this line

File: docker/docker-compose-check.sh [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
-main=`docker-compose  version  --short | cut -d '.' -f 1`
-minor=`docker-compose  version  --short | cut -d '.' -f 2`
-current=`docker-compose  version  --short`
+main=`docker compose  version  --short | cut -d '.' -f 1`
+minor=`docker compose  version  --short | cut -d '.' -f 2`
+current=`docker compose  version  --short`
 
-echo 'Checking docker-compose version'
+echo 'Checking docker compose version'
 if [[ $main -lt 1 ]]; then
```
Comment (dwalleck) [inline docker/docker-compose-check.sh:8]: Shouldn't this be `lt 2`? The error message below would also need to change

File: dc-build.sh (hunk 1 above)
Comment (trnayan) [reply dc-build.sh:17]: Hello @dwalleck, did you mean the comments line 16 & 17 both?

Comment (dwalleck) [reply dc-build.sh:17]: Yes, that was my thought. I could be talked into there being value in leaving the v1 compose commands for historical purposes.

**Comment by mtesauro** [comment]
@trnayan  I am good to approve once dwalleck's review items are handled.

File: dc-build.sh (hunk 1 above)
Comment (trnayan) [reply dc-build.sh]: Thanks for the observation. It's remove now.

**Review by dwalleck — APPROVED** [review]

**Review by cneill — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Pull request merged**